client/my-sites/customize/main.jsx

```diff
@@ -115,10 +123,32 @@ class Customize extends React.Component {
 		return false;
 	};
 
+	getReturnUrl = async () => {
+		const returnUrl = new URLSearchParams( window.location.search ).get( 'return' );
+
+		if ( ! returnUrl ) {
+			return null;
+		}
+
+		try {
+			const response = await wpcom.req.get( '/me/validate-redirect', { redirect_url: returnUrl } );
+
+			if ( ! response || ! response.redirect_to ) {
+				return null;
+			}
+
+			return response.redirect_to;
+		} catch {
+			// Ignore error, treat URL as invalid
+			return null;
+		}
+	};
+
 	getPreviousPath = () => {
 		let path = this.props.prevPath;
+
 		if ( ! path || /^\/customize\/?/.test( path ) ) {
-			path = '/stats';
+			path = this.props.isCustomerHomeEnabled ? '/home' : '/stats';
```

Comment:
Excellent change. One we missed when we were working on [this stuff](https://github.com/Automattic/wp-calypso/pull/39656)